File: hw/solutions/AC_Code.c
#include <stdio.h>
int main() {
    int k;
    

    scanf("%d", &k);

    for (int i = 0; i < k; i++) {
        unsigned int N;
        scanf("%u", &N);

        unsigned int even_bits = N & 0x55555555; // 0x55555555 = 01010101...0101
        unsigned int odd_bits = N & 0xAAAAAAAA;  // 0xAAAAAAAA = 10101010...1010

        even_bits <<= 1; 
        odd_bits >>= 1; 

        unsigned int result = even_bits | odd_bits;
        printf("%u\n", result);

    }

    return 0;
}

File: hw/solutions/AC_Code.py
k = int(input()) 
for _ in range(k):
    n = int(input())  
    arr = list(map(int, input().split()))  

    for i in range(n):
        N = arr[i]
        even_bits = N & 0x55555555  # 取出偶數位
        odd_bits = N & 0xAAAAAAAA   # 取出奇數位

        even_bits <<= 1  # 偶數位左移一位
        odd_bits >>= 1   # 奇數位右移一位

        result = even_bits | odd_bits
        print(result, end=" " if i != n - 1 else "\n")  
